plugins/filter/remove_keys.py

```diff
@@ -0,0 +1,154 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2024 Vladimir Botka <vbotka@gmail.com>
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = '''
+    name: remove_keys
+    short_description: Remove specific keys from dictionaries in a list.
+    version_added: "2.17"
+    author: Vladimir Botka (@vbotka)
+    description: This filter removes only specified keys from a provided list of dictionaries.
+    options:
+      _input:
+        description: A list of dictionaries.
+        type: list
+        elements: dictionary
+        required: true
+      target:
+        description:
+          - A list of keys or keys patterns to remove.
+          - The interpretation of the list items depends on the option C(matching_parameter)
+          - For matching_parameter C(regex) only the first item is taken.
+        type: list
+        elements: str
+        required: true
+      matching_parameter:
+        description: Specify the matching option of target keys.
+        type: str
+        default: equal
+        choices:
+          - equal
+          - starts_with
+          - ends_with
+          - regex
+'''
+
+EXAMPLES = '''
+  l:
+    - {k0_x0: A0, k1_x1: B0, k2_x2: [C0], k3_x3: foo}
+    - {k0_x0: A1, k1_x1: B1, k2_x2: [C1], k3_x3: bar}
+
+  # By default match equal keys.
+  t: [k0_x0, k1_x1]
+  r: "{{ l | remove_keys(target=t) }}"
+
+  # Match keys that starts with any of the items in the target.
+  t: [k0, k1]
+  r: "{{ l | remove_keys(target=t, matching_parameter='starts_with') }}"
+
+  # Match keys that ends with any of the items in target.
+  t: [x0, x1]
+  r: "{{ l | remove_keys(target=t, matching_parameter='ends_with') }}"
+
+  # Match keys by the regex.
+  t: ['^.*[01]_x.*$']
+  r: "{{ l | remove_keys(target=t, matching_parameter='regex') }}"
+
+  # Match keys by the regex. The regex does not have to be in list.
+  t: '^.*[01]_x.*$'
+  r: "{{ l | remove_keys(target=t, matching_parameter='regex') }}"
+
+  # The results of all examples are all the same.
+  r:
+    - {k2_x2: [C0], k3_x3: foo}
+    - {k2_x2: [C1], k3_x3: bar}
+'''
+
+RETURN = '''
+  _value:
+    description: The list of dictionaries with selected keys.
+    type: list
+    elements: dictionary
+'''
+
+from ansible.errors import AnsibleFilterError
+from ansible.module_utils.six import string_types
+from ansible.module_utils.common._collections_compat import Mapping, Sequence
+
+import re
+
+
+def remove_keys(data, target=None, matching_parameter='equal'):
+    """remove specific keys from dictionaries in a list"""
```

Comment:
The code here is very similar to the one in #8439. How about moving the common parts of the implementation to `plugins/plugin_utils/_keys_filter.py`, and only handle the differences in the filter plugin code?

Reply:
Thank you! I agree. I'll create a new PR.